### PR TITLE
Additions to the Coding Conventions and Pull Request sections in the Contribution Guidelines

### PR DESCRIPTION
**Pull request by Puddlestomper**
#### Proposed fix _(Make sure you've read [on how to contribute](https://github.com/TheCherno/Hazel/blob/master/.github/CONTRIBUTING.md) to Hazel)_
This PR expands on the coding conventions section of the contribution guidelines with regards to the naming scheme and adds points to the writing of a good PR.

The expansion of the coding conventions now lists all naming conventions.

The points added to "Writing a Good Pull Request" are:

- A reminder to be sure that submitted code compiles and passes all tests (when we do eventually set that up).
- A reminder to use closing keywords.
- A reminder to use descriptive commit messages.

There is also some general maintenance in the form of moving the links down to make the raw file more readable.

**EDIT:** Added more changes to the description above.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -50,11 +50,15 @@ If you're not sure what any of that means, check out Thinkful's [GitHub Pull Req
...
+- Make sure the code you submit compiles and runs without issues. When we set up unit tests and continuous integration we also expect that the pull request should pass all tests.
+
 - Follow our coding conventions, which we've intentionally kept quite minimal.
 
 ### Coding Conventions
 
-- For variables we use readable camel case: `doSomethingCool`. If they are class members use the 'm_' prefix: `m_DoSomethingCool`. When they are static use the 's_' prefix: `s_DoSomethingCool`.
+- For variables we use readable camel case: `variableName`. If they are class members use Hungarian notation (the 'm_' prefix): `m_ClassMemberVariableName`. When they are static use the 's_' prefix: `s_ClassStaticVariableName`. For class names we use Pascal case: `ClassName`. For functions we also use Pascal case: `FunctionName`, and we use camel case for parameters: `parameterName`.
+
+- For macros we use snake case: `MACRO_NAME` and if it is specifically related to Hazel we add the prefix 'HZ_' to it: `HZ_MACRO_NAME`.
```
Comment (LovelySanta) [inline .github/CONTRIBUTING.md]: This is nice, but I think this is a bit too long for one bullet point... And I think the macros we can add in the same bullet, as it is also about the naming?

- Naming convention:
    - For functions we use pascal case: **`FunctionName`**.
    - For (scoped) variables and function parameters we use camel case: **`variableName`** and **`parameterName`**.
    - For class names we use pascal case: **`ClassName`**.
    - For class variables we use the Hungarian notation:
        - Class member variables get the 'm_' prefix: **`m_ClassMemberVariableName`**.
        - Class static variables get the 's_' prefix: **`s_ClassStaticVariableName`**.
    - For macros we use snake case: **`MACRO_NAME`**.
        - If it is specifically related to Hazel, we add the 'HZ_' prefix: **`HZ_MACRO_NAME`**.
        - If there is a macro for the application and for the engine, we add an additional 'CORE_' prefix to the engine macro:  **`HZ_CORE_MACRO_NAME`**.

Comment (Puddlestomper) [reply .github/CONTRIBUTING.md]: Yes, this works much better. I've changed it as you suggested.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -52,6 +52,8 @@ If you're not sure what any of that means, check out Thinkful's [GitHub Pull Req
 
 - Make sure the code you submit compiles and runs without issues. When we set up unit tests and continuous integration we also expect that the pull request should pass all tests.
 
+- Use [closing keywords](https://help.github.com/en/articles/closing-issues-using-keywords)  in the appropriate section of our Pull Request template where applicable.
```
Comment (LovelySanta) [inline .github/CONTRIBUTING.md]: Not sure if it has impact... Just want to be sure.. And good work 👍 

```suggestion
- Use [closing keywords](https://help.github.com/en/articles/closing-issues-using-keywords) in the appropriate section of our Pull Request template where applicable.
```

Comment (Puddlestomper) [reply .github/CONTRIBUTING.md]: It's a good suggestion and I hope it will have an impact.

**Pull request merged**